src/textual/dom.py

```diff
@@ -280,8 +365,8 @@ def walk_children(self, with_self: bool = True) -> Iterable[DOMNode]:
                 pop()
             else:
                 yield node
-                if node.children:
-                    push(iter(node.children))
+                if node.node_list:
```

Comment:
NP: Was children perhaps a more semantic name?

Reply:
Yes, you're right! At one point I was going to split the nodes in to `children` and a `shadow node list`. I've put it back to `children`